pkg/operation/seed/seed.go

```diff
@@ -792,20 +781,54 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
 
 // DebootstrapCluster deletes certain resources from the seed cluster.
 func DebootstrapCluster(ctx context.Context, k8sSeedClient kubernetes.Interface) error {
+	bootstrapComponents, err := bootstrapComponents(k8sSeedClient, v1beta1constants.GardenNamespace, nil, nil)
+	if err != nil {
+		return err
+	}
+
 	// Delete component specific resources
 	var debootstrapFunctions []flow.TaskFn
-	for _, componentFn := range []component.DebootstrapSeed{
-		clusterautoscaler.DebootstrapSeed,
-		etcd.DebootstrapSeed,
-	} {
+	for _, componentFn := range bootstrapComponents {
+		fn := componentFn
 		debootstrapFunctions = append(debootstrapFunctions, func(ctx context.Context) error {
-			return componentFn(ctx, k8sSeedClient.Client(), v1beta1constants.GardenNamespace)
+			if err := fn.Destroy(ctx); err != nil {
+				return err
+			}
+			return fn.WaitCleanup(ctx)
```

Comment:
Alternatively, you can use:
```suggestion
			return component.OpDestroyAndWait(componentFn).Destroy(ctx)
```